packages/apollo-gateway/src/datasources/__tests__/RemoteGraphQLDataSource.test.ts

```diff
@@ -115,6 +119,50 @@ describe('willSendRequest', () => {
   });
 });
 
+describe('didReceiveResponse', () => {
+  it('can accept and modify context', async () => {
+    interface MyContext {
+      surrogateKeys: string[];
+    }
+
+    const DataSource = new RemoteGraphQLDataSource({
+      url: 'https://api.example.com/foo',
+      async didReceiveResponse(response, request, context) {
+        // I'd prefer to use super.didReceiveResponse here but it doesn't seem to work.
+        const body = await RemoteGraphQLDataSource.prototype.didReceiveResponse.call(
```

Comment:
Aha, `super` doesn't work because there is no super class of `RemoteGraphQLDataSource` when overriding the method from the constructor in the way that we area; `RemoteGraphQLDataSource` isn't an extension of anything, it only `implements GraphQLDataSource`:

https://github.com/apollographql/apollo-server/blob/e6ccd8bc41e51089ca428c408be2bcd0a8e3796d/packages/apollo-gateway/src/datasources/RemoteGraphQLDataSource.ts#L21

(So no `super`class!)

Reply:
Very helpful, thanks for pointing this out!
Resolved in 2792bdb